packages/runtime/lib/start.js

```diff
@@ -84,22 +75,30 @@ async function buildRuntime (configManager, env = process.env) {
       worker.postMessage({ signal: 'SIGUSR2' })
     })
 
-    // TODO(mcollina): refactor to not alter globals here
-    closeWithGrace((event, cb) => {
-      if (isWorkerAlive) {
-        worker.postMessage(event)
-        exited = cb
-      } else {
-        setImmediate(cb)
-      }
-    })
-
     /* c8 ignore next 3 */
     configManager.on('update', () => {
       // TODO(cjihrig): Need to clean up and restart the worker.
     })
   }
 
+  function setupExit () {
+    worker.on('exit', (code) => {
+      if (exiting || !runtimeApiClient.started) {
```

Comment:
See another comment ^^^. I don't really understand the usecase for the second condition in this if.

Reply:
I've added a comment to explain the second condition.